arrow-row/src/lib.rs

```diff
@@ -1951,11 +1956,26 @@ unsafe fn decode_column(
 
                         let child_array =
                             unsafe { converter.convert_raw(&mut sparse_data, validate_utf8) }?;
+
+                        // track bytes consumed for rows that belong to this field
+                        for (row_idx, child_row) in field_rows.iter() {
+                            let remaining_len = sparse_data[*row_idx].len();
+                            bytes_consumed[*row_idx] = 1 + child_row.len() - remaining_len;
+                        }
```

Comment:
```suggestion
                        // ensure we advance pass consumed bytes in rows
                        for (row_idx, child_row) in field_rows.iter() {
                            let remaining_len = sparse_data[*row_idx].len();
                            let consumed_length = 1 + child_row.len() - remaining_len;
                            rows[*row_idx] = &rows[*row_idx][consumed_length..];
                        }
```

Thoughts of inlining it like this, which can remove the need for a separate `bytes_consumed` vec?

Reply:
I like it!